presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -868,57 +870,38 @@ public static Slice stEndPoint(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     @SqlType("array(" + GEOMETRY_TYPE_NAME + ")")
     public static Block stPoints(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     {
-        OGCGeometry geometry = deserialize(input);
+        Geometry geometry = deserialize(input);
         validateType("ST_Points", geometry, EnumSet.of(LINE_STRING, POLYGON, POINT, MULTI_POINT,
                 MULTI_LINE_STRING, MULTI_POLYGON, GEOMETRY_COLLECTION));
         if (geometry.isEmpty()) {
             return null;
         }
 
-        int pointCount = countNumPoints(geometry);
+        int pointCount = geometry.getNumPoints();
         BlockBuilder blockBuilder = GEOMETRY.createBlockBuilder(null, pointCount);
         buildPointsBlock(geometry, blockBuilder);
-        return blockBuilder.build();
-    }
-
-    private static int countNumPoints(OGCGeometry geometry)
-    {
-        if (geometry == null) {
-            return 0;
-        }
 
-        GeometryType type = GeometryType.getForEsriGeometryType(geometry.geometryType());
-        int pointCount = 0;
-        if (type == GeometryType.GEOMETRY_COLLECTION) {
-            OGCGeometryCollection collection = (OGCGeometryCollection) geometry;
-            for (int i = 0; i < collection.numGeometries(); i++) {
-                OGCGeometry entry = collection.geometryN(i);
-                pointCount += countNumPoints(entry);
-            }
-        }
-        else {
-            pointCount = getPointCount(geometry);
-        }
-        return pointCount;
+        return blockBuilder.build();
     }
 
-    private static void buildPointsBlock(OGCGeometry geometry, BlockBuilder blockBuilder)
+    private static void buildPointsBlock(Geometry geometry, BlockBuilder blockBuilder)
     {
-        GeometryType type = GeometryType.getForEsriGeometryType(geometry.geometryType());
+        GeometryType type = GeometryType.getForEsriGeometryType(geometry.getGeometryType());
```

Comment:
If this is a JTS geometry now, this should use `getForJtsGeometryType`

Reply:
Got it.